src/timeout.c

```diff
@@ -54,18 +55,24 @@ int checkBlockedClientTimeout(client *c, mstime_t now) {
  * each iteration would be costly without any actual gain. */
 int clientsCronHandleTimeout(client *c, mstime_t now_ms) {
     time_t now = now_ms/1000;
-
+    int drop_client = 0;
     if (server.maxidletime &&
         /* This handles the idle clients connection timeout if set. */
         !(c->flags & CLIENT_SLAVE) &&   /* No timeout for slaves and monitors */
         !mustObeyClient(c) &&         /* No timeout for masters and AOF */
         !(c->flags & CLIENT_BLOCKED) && /* No timeout for BLPOP */
         !(c->flags & CLIENT_PUBSUB) &&  /* No timeout for Pub/Sub clients */
-        (now - c->lastinteraction > server.maxidletime))
+        (now - c->lastinteraction > server.maxidletime)) 
     {
-        serverLog(LL_VERBOSE,"Closing idle client");
-        freeClient(c);
-        return 1;
+        drop_client = 1;
+    } else if (c->authenticated == 0 &&
+               now - c->ctime > MAX_TIME_TO_AUTH &&
+               !(c->flags & (CLIENT_CLOSE_AFTER_COMMAND | CLIENT_CLOSE_AFTER_REPLY | CLIENT_CLOSE_ASAP))) 
+    {
+        /* Limit time of client authentication and exclude 
+         * CLIENT_CLOSE_AFTER_COMMAND, CLIENT_CLOSE_AFTER_REPLY, 
+         * CLIENT_CLOSE_ASAP*/
```

Comment:
somehow i feel the comment is a bit redundant. since we do list these flags in the statement
```suggestion
        /* Limit time of client authentication and exclude some flags. */
```